array-manipulation-3/src/flatten.js

```diff
@@ -0,0 +1,26 @@
+/* exported flatten */
+function flatten(array) {
+  var flattenedArray = [];
+  for (var i = 0; i < array.length; i++) {
+    if (Array.isArray(array[i]) === true) {
+      var arrayOfArray = array[i];
+      for (var n = 0; n < arrayOfArray.length; n++) {
+        flattenedArray.push(arrayOfArray[n]);
+      }
+    } else if ((typeof array[i] === 'number') || (typeof array[i] === 'boolean')) {
+      flattenedArray.push(array[i]);
+    }
```

Comment:
I think just an `else` should work here, no? You want your algorithm to be general and not skip other data types.